lib/ui/window/platform_configuration.h

```diff
@@ -0,0 +1,126 @@
+  }
+};
+
+}  // namespace tonic
+
+namespace flutter {
+class FontCollection;
+class PlatformMessage;
+class Scene;
+
+// Must match the AccessibilityFeatureFlag enum in framework.
+enum class AccessibilityFeatureFlag : int32_t {
+  kAccessibleNavigation = 1 << 0,
+  kInvertColors = 1 << 1,
+  kDisableAnimations = 1 << 2,
+  kBoldText = 1 << 3,
+  kReduceMotion = 1 << 4,
+  kHighContrast = 1 << 5,
+};
+
+class PlatformConfigurationClient {
+ public:
+  virtual std::string DefaultRouteName() = 0;
+  virtual void ScheduleFrame() = 0;
+  virtual void Render(Scene* scene) = 0;
+  virtual void UpdateSemantics(SemanticsUpdate* update) = 0;
+  virtual void HandlePlatformMessage(fml::RefPtr<PlatformMessage> message) = 0;
+  virtual FontCollection& GetFontCollection() = 0;
+  virtual void UpdateIsolateDescription(const std::string isolate_name,
+                                        int64_t isolate_port) = 0;
+  virtual void SetNeedsReportTimings(bool value) = 0;
+  virtual std::shared_ptr<const fml::Mapping> GetPersistentIsolateData() = 0;
+  virtual std::unique_ptr<std::vector<std::string>>
+  ComputePlatformResolvedLocale(
+      const std::vector<std::string>& supported_locale_data) = 0;
+
+ protected:
+  virtual ~PlatformConfigurationClient();
+};
+
+class PlatformConfiguration final {
+ public:
+  explicit PlatformConfiguration(PlatformConfigurationClient* client);
+
+  ~PlatformConfiguration();
+
+  PlatformConfigurationClient* client() const { return client_; }
+
+  void DidCreateIsolate();
+  void UpdateLocales(const std::vector<std::string>& locales);
+  void UpdateUserSettingsData(const std::string& data);
+  void UpdateLifecycleState(const std::string& data);
+  void UpdateSemanticsEnabled(bool enabled);
+  void UpdateAccessibilityFeatures(int32_t flags);
+  void DispatchPlatformMessage(fml::RefPtr<PlatformMessage> message);
+  void DispatchPointerDataPacket(const PointerDataPacket& packet);
+  void DispatchSemanticsAction(int32_t id,
+                               SemanticsAction action,
+                               std::vector<uint8_t> args);
+  void BeginFrame(fml::TimePoint frameTime);
+  void ReportTimings(std::vector<int64_t> timings);
+
+  void CompletePlatformMessageResponse(int response_id,
+                                       std::vector<uint8_t> data);
+  void CompletePlatformMessageEmptyResponse(int response_id);
+
+  static void RegisterNatives(tonic::DartLibraryNatives* natives);
+
+  //----------------------------------------------------------------------------
+  /// @brief Retrieves the Window.
+  ///
+  /// @return the Window.
+  const Window& window() const { return *window_; }
+
+  //----------------------------------------------------------------------------
+  /// @brief Sets the viewport metrics of the Window.
+  ///
+  /// @param[in] window_metrics The viewport metrics to replace the old ones
+  ///            with.
+  void SetWindowMetrics(const ViewportMetrics& window_metrics);
```

Comment:
Shouldn't this just be a Window method? I'm not sure why we need the passthrough.